src/software/simulated_tests/non_terminating_validation_functions/robots_not_in_center_circle_validation.cpp

```diff
@@ -3,21 +3,32 @@
 #include <gtest/gtest.h>
 
 #include "software/geom/algorithms/contains.h"
+#include "software/logger/logger.h"
 
 void robotsNotInCenterCircle(std::shared_ptr<World> world_ptr,
                              ValidationCoroutine::push_type& yield)
 {
     for (auto robot : world_ptr->friendlyTeam().getAllRobots())
     {
-        robotNotInCenterCircle(robot, world_ptr, yield);
+        if (contains(world_ptr->field().centerCircle(), robot.position()))
+        {
+            FAIL() << "Robot " + std::to_string(robot.id()) + " entered center circle";
+        }
```

Comment:
why can't this use robotNotInCenterCircle?

Reply:
It can but `robotNotInCenterCircle` takes in a RobotId and finds a robot based on that, but in `robotsNotInCenterCircle`, it already has the robots, so we don't need to find the RobotId. I think it would be redundant if we use `robotNotInCenterCircle` since I would have to get the robot_id from a robot -> pass this into `robotNotInCenterCircle`, when I already have the robot found already.